src/traces/histogram/attributes.js

```diff
@@ -71,6 +71,16 @@ module.exports = {
         ].join(' ')
     },
 
+    mode: {
```

Comment:
This is a cool function--I think I'd prefer a `cumulative` boolean attribute. That's pretty explicit and +1 about not seeing any other possible *modes*.
Also, not sure if this is being too nitpicky across traces, but I'm not super convinced that this use case provides a good parallel to [scatter mode](https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/attributes.js#L85)